File: cpp/101-symmetric-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool checkSymmetric(TreeNode* node1, TreeNode* node2){
        if(node1 == nullptr && node2 == nullptr)
            return true;
        else if(node1 == nullptr || node2 == nullptr)
            return false;
        if(node1->val != node2->val)
            return false;
        return checkSymmetric(node1->left, node2->right) && checkSymmetric(node1->right, node2->left);
    }
    
    bool isSymmetric(TreeNode* root) {
       return checkSymmetric(root, root);
    }
};

File: py/101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        q = []
        q.append(root)
        q.append(root)
        
        while q:
            n1 = q.pop()
            n2 = q.pop()
            if not n1 and not n2:
                continue
            elif not n1 or not n2:
                return False
            elif n1.val != n2.val:
                return False
            q.append(n1.left)
            q.append(n2.right)
            q.append(n1.right)
            q.append(n2.left)
        
        return True